### ISSUE TITLE
Unable to create users with the API

### ISSUE THREAD

**Issue by stefannitsche** [grokability/snipe-it#8986]
#### Please confirm you have done the following before posting your bug report:

- [x] I have enabled debug mode 
- [x] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

**Describe the bug**
It is not possible to create new user through the API. **Please note:** This is not related to a missing or incorrect API-key. With the same API-key it's possible to create companies, assets etc, only user creation is affected.

**To Reproduce**
1. Create a brand new installation using the official Docker image
2. Log in and create an api key
3. curl --include --request POST --url http://localhost:3051/api/v1/users --header 'Authorization: Bearer <api_key>' --header 'Content-Type: application/json' --data '{"first_name":"MrTest","last_name":"WillIFail","username":"api_fail_tester","password":"Mhunamana!","email":"api.fail@example.com","activated":false}'
4. Result: Status code 302, a redirection to the root of the site and no user created (**Note:** A missing API-key will not give the same result, it will give status code 302 and redirect to /login)

**Expected behavior**
Status code 200, a json object confirming the creation of the user and the user being created.

**Screenshots**
https://user-images.githubusercontent.com/13170419/104030026-09d38480-51cb-11eb-9ce9-6462d2b40c51.mp4

**Server (please complete the following information):**
 - Snipe-IT Version: 5.0.12
 - OS: Official Docker Image
 - Web Server: Official Docker Image
 - PHP Version: Official Docker Image

**Desktop (please complete the following information):**
 - OS: Ubuntu on WSL
 - Browser curl
 - Version 7.68.0

**Error Messages**
- Debug turned on doesn't yield any extra information.
- API feature is turned off on https://snipeitapp.com/demo
- The issue is reproducable on https://snipe-it.readme.io/reference#users-2

**Additional context**
- Fresh install and upgrades (both have been tested)
- Offical Docker image
- Docker
- Follow the installation guide, login as super admin, create API-key, run curl to create user and the issue appears.
- No direct manipulation of the database

I've previously tried to raise this issue (#8784) but that was unfortunatley closed with the appearence that some parts of the report was ignored or not read. Hopefully all information I've provided will be taken into account this time.

**Comment by albinbatman** [comment]
Hi @stefannitsche,

You are missing one important header and without it the application returns a 302. Unfortunately the test that you mentioned in your previous issue, does not state this information.

Please try your curl request with the following header attached:

```
--header 'Accept: application/json'
```

Also, with the data you have right now it will fail as I see because you must also provide "password_confirmation" in the json object (also not seen in the test):
```
{
   "first_name":"test",
   "username":"test",
   "password":""eightcharacterslong",
   "password_confirmation":"eightcharacterslong"
}
```

Here is example curl request:

```bash
Albin@localhost MINGW64 ~
$ curl --include --request POST --url http://<your domain>/api/v1/users --header 'Authorization: Bearer <your api key>' --header 'Content-Type: application/json' --header 'Accept: application/json' --data '{"first_name":"test","username":"test","password":"eightcharacterslong","password_confirmation":"eightcharacterslong"}'
```

example server reply:

```
HTTP/1.1 200 OK
Server: nginx/1.16.0
Content-Type: application/json
Transfer-Encoding: chunked
Connection: keep-alive
Cache-Control: no-cache, private
Date: Sat, 09 Jan 2021 06:29:38 GMT
X-Clockwork-Id: X9dc12e3b47bf28520a2e3ce8a2e5df67
X-Clockwork-Version: 1
X-Clockwork-Path: _debugbar/clockwork/
phpdebugbar-id: X9dc12e3b47bf28520a2e3ce8a2e5df67
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Feature-Policy: accelerometer 'none';ambient-light-sensor 'none';animations 'none';autoplay 'none';battery 'none';camera 'none';display-capture 'none';document-domain 'none';encrypted-media 'none';fullscreen 'none';geolocation 'none';gyroscope 'none';legacy-image-formats 'none';magnetometer 'none';microphone 'none';midi 'none';oversized-images 'none';payment 'none';picture-in-picture 'none';publickey-credentials 'none';sync-xhr 'none';unsized-media 'none';usb 'none';vibrate 'none';wake-lock 'none';xr-spatial-tracking 'none'
Referrer-Policy: same-origin
X-Frame-Options: DENY
Set-Cookie: snipeit_session=k3M8lTekVqPin6aQJRlA0cn7MX00a2umOEAUQ6AB; expires=Sun, 17-Jan-2021 14:29:38 GMT; Max-Age=720000; path=/; httponly

{"status":"success","messages":"User was successfully created.","payload":{"id":63,"avatar":"http:\/\/snipe-it.test\/img\/default-sm.png","name":"test ","first_name":"test","last_name":"","username":"test","employee_num":"","manager":null,"jobtitle":null,"phone":null,"website":null,"address":null,"city":null,"state":null,"country":null,"zip":null,"email":"","department":null,"location":null,"notes":"","permissions":null,"activated":false,"ldap_import":false,"two_factor_activated":false,"two_factor_enrolled":false,"assets_count":0,"licenses_count":0,"accessories_count":0,"consumables_count":0,"company":null,"created_at":{"datetime":"2021-01-09 06:29:38","formatted":"Sat Jan 09, 2021 6:29AM"},"updated_at":{"datetime":"2021-01-09 06:29:38","formatted":"Sat Jan 09, 2021 6:29AM"},"last_login":null,"deleted_at":null,"available_actions":{"update":true,"delete":true,"clone":true,"restore":false},"groups":null}}
```

**Comment by stefannitsche** [comment]
Hi @albinbatman,

Thank you very much for your reply. Adding the extra header and the password_confirmation field does indeed make it possible to create users. But it also exposes the reason behind my issue.

While looking up how to use the API I ended up on https://snipe-it.readme.io/reference which, as far as I can tell, seems to be the offical documentation for the api. There I looked up /users and more specifically the POST request to /users ( https://snipe-it.readme.io/reference#users-2 ). When constructing my payload I used the list of body params as a guide and that list doesn't even mention the existence of "password_confirmation". This makes it kind of hard to know that a field not even mentioned is a required field.

This also leads in to how I constructed my query (since I was missing a header). Yet again I used https://snipe-it.readme.io/reference#users-2 as my template and here is how the sample query looks in the documentation:

```
curl --request POST \
  --url https://develop.snipeitapp.com/api/v1/users \
  --header 'Authorization: Bearer <api key>' \
  --header 'Content-Type: application/json' \
  --data '{"first_name":"test","username":"test","password":"eightcharacterslong"}'
```

Above example is directly copied from the documentation the only modification I've done is that I removed the API-key for readability. Here there is no mentioning of the need of a third header for accepting json. Yet again it is kind of hard to include information you aren't aware of.

So while there doesn't seem to be an issue with the API per se there seems to be an issue with the documentation of the API. If https://snipe-it.readme.io/reference is the intended place to learn about the API it should be updated to actually contain the needed information. If it isn't the intended place it should preferable be removed or at least not be linked to. I ended up there following the link on this page: https://snipe-it.readme.io/docs/internally-parsing-the-api

**TL;DR** API seems to work fine when everything needed is included but the documentation doesn't seem to include everything you need to know to make working requests.

<blockquote><img src="/favicon.ico" width="48" align="right"><div>Snipe-IT Documentation</div><div><strong><a href="https://snipe-it.readme.io/reference">Overview</a></strong></div><div>Snipe-IT is a free, open source IT asset management system. Features include management of assets, users, licenses, accessories, consumables and components, as well as two-factor authentication, LDAP/AD syncing, and asset acceptance confirmation.</div></blockquote>
<blockquote><img src="/favicon.ico" width="48" align="right"><div>Snipe-IT Documentation</div><div><strong><a href="https://snipe-it.readme.io/reference#users-2">Overview</a></strong></div><div>Snipe-IT is a free, open source IT asset management system. Features include management of assets, users, licenses, accessories, consumables and components, as well as two-factor authentication, LDAP/AD syncing, and asset acceptance confirmation.</div></blockquote>
<blockquote><img src="/favicon.ico" width="48" align="right"><div>Snipe-IT Documentation</div><div><strong><a href="https://snipe-it.readme.io/docs/internally-parsing-the-api">Internally Parsing the API</a></strong></div><div>Snipe-IT uses the fantastic (if roughly documented) Bootstrap-Table library from wenzhixin to handle all list views in the web GUI. Starting in version 4.0 of Snipe-IT, we've dramatically changed how the JSON responses generated by Snipe-IT are consumed. Version 4.0 cleans up the JSON, and offloads ...</div></blockquote>

**Comment by snipe** [comment]
The curl example on the right side of the example states:

```
curl --request POST \
     --url https://develop.snipeitapp.com/api/v1/users \
     --header 'Accept: application/json' \
     --header 'Authorization: Bearer <api key>' \
     --header 'Content-Type: application/json'
```
